[3851] Sidebar Browse Widget

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/3851

aachavarria committed:
fixes on `import makeStyles from '@material-ui/styles/makeStyles'` import;
adding PagesWidget;

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,145 @@
+    <section className={classes.wrapper}>
+      <PagesHeader/>
+      <PagesBreadcrumbs/>
+      <PagesNav/>
```
Like this breakdown. Just missing pagination.

aachavarria committed:
renaming item to legacyItem;
adding getChildrenByPath service;
pagesWidget: onItemSelected(), items Map, etc;

aachavarria committed:
adding pagination;

aachavarria committed:
adding colors; adding menus; Breadcrumbs changes;

aachavarria committed:
breadcrumbs changes;
select/unselect;
locale support;

aachavarria committed:
pagesWidget: menu refactor;

rart commented on ui/app/src/components/CustomMenu.tsx:
```diff
@@ -0,0 +1,77 @@
+    <Menu
+      anchorEl={anchorEl}
+      open={open}
+      classes={{ paper: classes.paper }}
```
pass down classes

rart commented on ui/app/src/components/CustomMenu.tsx:
```diff
@@ -0,0 +1,77 @@
+    >
+      {
+        sections.map((section: any, i: number) =>
+          section.map((SectionItem: SectionItem, y: number) =>
```
begin with lowercase

rart commented on ui/app/src/components/CustomMenu.tsx:
```diff
@@ -0,0 +1,77 @@
+    >
+      {
+        sections.map((section: any, i: number) =>
+          section.map((SectionItem: SectionItem, y: number) =>
```
Sure this works? You're still returning array of arrays.
Might need to use flatMap instead

aachavarria replied:
works very good (Y)

rart commented on ui/app/src/components/CustomMenu.tsx:
```diff
@@ -0,0 +1,77 @@
+              ) : null
+            ) : (
+              <MenuItem
+                divider={!!((i !== sections.length - 1) && (y === section.length - 1))}
```
unneeded boolean coercing

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+  };
+
+  const changeLocale = (locale: string) => {
+    setMenu({ ...menu, anchorEl: null });
```
useReducer. if props don't change often, separate the state items.
Then, change all the setMenu to set only the prop you want to change.

aachavarria replied:
they almost always change together

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+              ) :
+              (
+                <Typography key={item.id} variant="subtitle2"
+                            className={classes.pagesBreadcrumbsTypo}>{item.label}</Typography>
```
Format!

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+        over &&
+        <div className={classes.optionsWrapper}>
+          <IconButton aria-label="options" className={classes.icon}
+                      onClick={(event) => onOpenItemMenu(event.currentTarget)}>
```
Format

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+  useEffect(() => {
+    getChildrenByPath(site, activePath).subscribe(
+      (response) => {
+        console.log(response);
```
x

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+        setItems(response.items);
+      },
+      (error) => {
+        console.log(error);
```
Handle

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+    <header className={classes.pagesHeader}>
+      <DescriptionOutlinedIcon/>
+      <Typography variant="subtitle1" className={classes.pagesHeaderTitle}>
+        Pages
```
Translate

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+          ) : (
+            <MenuItem onClick={() => {
+            }}>
+              option1
```
?

aachavarria replied:
this menu does not have an options mockup on figma

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+  };
+
+  const onPageChanged = (page: number) => {
+    console.log(page);
```
x

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,692 @@
+      id: 'Home',
+      label: 'Home',
+      path: 'home'
+    }
```
This should also come from the API response. Re-check the API specs for updates.

aachavarria replied:
the api is not returning any breadcrumb

rart replied:
Not breadcrumb but parent item is returned too. Review the updated API response

aachavarria committed:
Pr requested changes;

aachavarria committed:
adding items label to select;
adding error State;

aachavarria committed:
pagesWidget: adding searchBar;
adding SearchBar, delete palette after merge; adding persistentIcon prop;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3851

aachavarria committed:
CustomMenu.tsx deleting validation;
PagesWidget.tsx adding copy and paste, changes on generate menu;
searchBar.tsx removing palette;
adding copyItem interface;
adding new services;
addings hooks.ts;

aachavarria committed:
adding utils/object.ts
adding models/GlobalState.ts
adding errorHandling and cutItem;
adding errorDialog, errorState;

aachavarria committed:
adding copyDialog to pagesWidget;
adding children to legacyItem;
adding new services;
updating material-ui/lab;

aachavarria committed:
adding props, and format;

aachavarria committed:
onFocus issue

aachavarria committed:
using palette on pagesWidget;
extracting Dialogtitle;
Adding ContentLocalizationDialog.tsx;

aachavarria committed:
translation.ts: adding service getTargetLocales;
item.ts: adding interface localizationItem;
pagesWidget.tsx: adding translations and contentLocalizationDialog;
adding support to translations on CustomMenu;
contentLocalizationDialog.tsx: adding translations;

aachavarria committed:
deleting useEffect

aachavarria committed:
adding markForTranslation service;
error handling;

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+          },
+          ({ response }) => {
+            //TODO: I'm wrapping the API response as a API2 response
+            const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
We should do this at the service level

aachavarria replied:
removing because markForTranslation is api2

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+  toggleSelectAll(): void;
+}
+
+function ItemHeaderOptions(props: ItemHeaderOptionsProps) {
```
We need this as a reusable component (on its own).

Rename to ContextSelectionOptionsOverlay

aachavarria replied:
should move this to a .tsx file?

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+            color="inherit"
+            component="button"
+            variant="subtitle2"
+            TypographyClasses={{ root: classes.linkTypo }}
```
linkTypo => optionTypography

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+  };
+
+  const onOptionClicked = (option: Option) => {
+    console.log(option);
```
x

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+                />
+                <>
+                  <Typography variant="subtitle2" className={clsx(classes.headerTitle, classes.width30)}>
+                    {formatMessage(translations.locales)}
```
You could use FormattedMessage component form in most of the usages here

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+import Dialog from '@material-ui/core/Dialog';
+import { defineMessages, useIntl } from 'react-intl';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import { Typography } from '@material-ui/core';
```
Import from its individual file

rart replied:
👆🏻

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,393 @@
+import { palette } from '../styles/theme';
+import Checkbox from '@material-ui/core/Checkbox';
+import IconButton from '@material-ui/core/IconButton';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
```
Use rounded icons

rart commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+  return (
+    <MuiDialogTitle disableTypography className={classes.titleRoot}>
+      <div className={classes.title}>
+        <Typography variant="h6">{title}</Typography>
```
add props (with default values) for titleTypographyProps, subtitleTypographyProps. Put variants in there.

aachavarria replied:
will handle it on the anotherPR

rart commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+  onClose(): void;
+}
+
+export default function DialogTitle(props: DialogTitleProps) {
```
Rename file and component to DialogHeader

aachavarria replied:
renaming the other ones on BulkUpload PR, because that one has the updates

rart commented on ui/app/src/components/SearchBar.tsx:
```diff
@@ -0,0 +1,113 @@
+      />
+      {
+        (closeIcon && (persistentCloseIcon || keyword)) &&
+        <CloseIcon className={classes.closeIcon} onClick={() => onChange('')}/>
```
- Add icon button component to wrap the icon, 
- make the button handler call a custom prop called onActionButtonClick, 
- rename closeIcon to showActionButton - this could be a function or a boolean 
- add actionButtonIcon prop for configurable action
- remove persistentCloseIcon to persistentActionButton - in favour of showActionButton

rart commented on ui/app/src/components/SearchBar.tsx:
```diff
@@ -0,0 +1,113 @@
+  const { formatMessage } = useIntl();
+  return (
+    <div className={clsx(classes.search, focus && 'focus', props.classes?.root)}>
+      <SearchIcon className={classes.searchIcon}/>
```
- add showDecoratorIcon: boolean prop with default to true
- add decoratorIcon prop with default to SearchIcon(Rounded)

rart commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+  onClose(): void;
+}
+
+export default function DialogTitle(props: DialogTitleProps) {
```
Rename file and component to DialogHeader

rart commented on ui/app/src/components/DialogTitle.tsx:
```diff
@@ -0,0 +1,71 @@
+import { palette } from '../styles/theme';
+import MuiDialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
```
Rounded

aachavarria replied:
it was changed on https://github.com/craftercms/studio-ui/pull/1638

rart commented on ui/app/src/components/SystemStatus/ErrorDialog.tsx:
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { APIError } from '../../models/GlobalState';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
```
Rounded

rart commented on ui/app/src/services/content.ts:
```diff
@@ -30,6 +31,62 @@ export function getDOM(site: string, path: string): Observable<XMLDocument> {
   );
 }
 
+export function getChildrenByPath(site: string, path: string): Observable<any> {
```
Add new fns to default export object below

rart commented on ui/app/src/components/CustomMenu.tsx:
```diff
@@ -0,0 +1,84 @@
+/*
```
Let’s rename to context menu

rart commented on ui/app/src/components/CopyItemsDialog.tsx:
```diff
@@ -0,0 +1,298 @@
+        <DialogTitle title={title} subtitle={subtitle} onClose={onClose}/>
+      }
+      <DialogContent dividers classes={{ root: classes.dialogContentRoot }}>
+        <SimpleItemsSelection
```
=> ItemSelectorTree

rart commented on ui/app/src/components/SystemStatus/ErrorDialog.tsx:
```diff
@@ -0,0 +1,58 @@
+}
+
+export default function ErrorDialog(props: ConfirmDialogProps) {
+  const { onClose, error } = props;
```
Add errorStateProps and spread them in the error state
rename props interface

rart commented on ui/app/src/components/SystemStatus/ErrorDialog.tsx:
```diff
@@ -0,0 +1,58 @@
+      <IconButton
+        aria-label="close"
+        className={classes.closeButton}
+        onClick={() => onClose()}
```
onClick={onClose}

aachavarria committed:
PR feedback, renaming,etc;

aachavarria committed:
fixes on menu sections

aachavarria committed:
updates on breadcrumbs

aachavarria committed:
ContentLocalizationDialog: changes on imports for makestyles/createstyles

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/yarn.lock

aachavarria committed:
updating materialUI

aachavarria committed:
using Rounded Icons

aachavarria committed:
SearchBar changes, removing persistentCloseIcon, using showActionButton={true} or showActionButton{keyword}.

rart commented on ui/app/src/components/BlueprintCard.tsx:
```diff
@@ -15,23 +15,23 @@
+import MobileStepper from './MobileStepper';
+import { backgroundColor } from '../styles/theme';
+import Button from '@material-ui/core/Button';
+import { Theme, Tooltip } from '@material-ui/core';
```
Import individual files from /styles and /core/Tooltip

rart commented on ui/app/src/components/CopyItemsDialog.tsx:
```diff
@@ -0,0 +1,298 @@
+import React, { useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+import makeStyles from '@material-ui/styles/makeStyles';
+import { Link, Theme } from '@material-ui/core';
```
Import individual files

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+                },
+                ({ response }) => {
+                  //TODO: I'm wrapping the API response as a API2 response
+                  const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
Move this abstraction to the service

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+          },
+          ({ response }) => {
+            //TODO: I'm wrapping the API response as a API2 response
+            const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
Move this abstraction to the service

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+            setMenuState({ hasClipboard: false });
+          },
+          ({ response }) => {
+            const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
To service

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+          },
+          ({ response }) => {
+            //TODO: I'm wrapping the API response as a API2 response
+            const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
To service

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+          },
+          ({ response }) => {
+            //TODO: I'm wrapping the API response as a API2 response
+            const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
```
To service

rart commented on ui/app/src/components/PagesWidget.tsx:
```diff
@@ -0,0 +1,1023 @@
+      ({ response }) => {
+        //TODO: I'm wrapping the API response as a API2 response
+        const error = { ...response, code: '', documentationUrl: '', remedialAction: '' };
+        setError(error);
```
To service

rart commented:
Pull & merge conflicts

aachavarria committed:
changes on the imports;

aachavarria committed:
changes on the imports;
adding util catchApi1Error, to wrap api1 response;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/src/services/content.ts
	ui/app/src/utils/hooks.ts

aachavarria committed:
removing merge issue

rart commented on ui/app/src/components/BlueprintCard.tsx:
```diff
@@ -15,23 +15,24 @@
+import Button from '@material-ui/core/Button';
+import Tooltip from '@material-ui/core/Tooltip';
+import clsx from 'clsx';
+import { Theme } from '@material-ui/core';
```
👆🏻

aachavarria replied:
Theme dont have default export

aachavarria replied:
changed

rart commented on ui/app/src/components/BlueprintCard.tsx:
```diff
@@ -289,6 +290,7 @@ function BlueprintCard(props: BlueprintCardProps) {
         {
           (id !== 'GIT') &&
           <CardActionArea onClick={() => onBlueprintSelected(blueprint, 1)}>
+            // @ts-ignore
```
```typescript
{/*
// @ts-ignore */}
```

rart commented on ui/app/src/components/ContentLocalizationDialog.tsx:
```diff
@@ -0,0 +1,391 @@
+    }
+  };
+
+  const onOptionClicked = (option: Option) => {
```
Does this have a purpose?

aachavarria replied:
yes

aachavarria committed:
Merge branch '2019-11-ui-restructure' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/package.json
	ui/app/src/components/DependencySelection.tsx
	ui/app/src/components/SearchBar.tsx
	ui/app/src/components/SystemStatus/ErrorDialog.tsx
	ui/app/src/models/GlobalState.ts
	ui/app/src/modules/System/Sites/Create/BlueprintCard.tsx
	ui/app/src/services/content.ts
	ui/app/src/utils/ajax.ts
	ui/app/src/utils/object.ts
	yarn.lock

aachavarria committed:
resolving merge conflicts, adding changes on SearchBar

aachavarria committed:
adding comment

rart reviewed (changes requested):
- Conflicts
- Colocate publish dialog components on the PublishDialog.tsx
- Review PRs with modifications to files here to assess conflicts

aachavarria committed:
Merge branch '2019-11-ui-restructure' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/src/components/SystemStatus/ErrorDialog.tsx
	ui/app/src/modules/Preview/Tools/AssetsPanel.tsx
	ui/app/src/modules/Preview/Tools/BrowseComponentsPanel.tsx
	ui/app/src/modules/System/Sites/Create/CreateSiteDialog.tsx

aachavarria committed:
Prettier;
Using dispatch to show the ErrorDialog;

aachavarria committed:
Merge branch '2019-11-ui-restructure' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/src/services/content.ts
	ui/app/src/utils/ajax.ts

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3851

 Conflicts:
	ui/app/src/components/SearchBar.tsx
	ui/app/src/modules/Search/Search.tsx
